File: 2373/solution.java
class Solution {
    public int[][] largestLocal(int[][] grid) {
        int N = grid.length;
        int[][] res = new int[N - 2][N - 2];
        for (int row = 0; row < N - 2; row++) {
            for (int col = 0; col < N - 2; col++) {
                for (int i = -1; i <= 1; i++) {
                    for (int j = -1; j <= 1; j++) {
                        res[row][col] = Math.max(res[row][col], grid[row + 1 + i][col + 1 + j]);
                    }
                }
            }
        }
        return res;
    }
}

// time O(n^2)
// space O(1)

File: 2373/solution.py
class Solution:
    def largestLocal(self, grid: list[list[int]]) -> list[list[int]]:
        N = len(grid)
        res = [[0 for col in range(N - 2)] for row in range(N - 2)]
        for row in range(N - 2):
            for col in range(N - 2):
                for i in range(-1, 2):
                    for j in range(-1, 2):
                        res[row][col] = max(
                            res[row][col], grid[row + i + 1][col + j + 1]
                        )
        return res
